pkgs/development/compilers/llvm/common/libcxx/default.nix

```diff
@@ -63,6 +63,8 @@ let
     "-DLIBCXXABI_ENABLE_EXCEPTIONS=OFF"
   ] ++ lib.optionals (!enableShared) [
     "-DLIBCXXABI_ENABLE_SHARED=OFF"
+  ] ++ lib.optionals stdenv.hostPlatform.isStatic [
+    "-DLIBCXX_STATICALLY_LINK_ABI_IN_STATIC_LIBRARY=On"
```

Comment:
What do you think about _not_ gating this on `hostPlatform.isStatic`?

I know using `cc` from `!isStatic` stdenvs for static linking isn't a supported configuration but I think this still ends up happening for some out-of-nixpkgs use cases (i.e. `devShell`s) and we _do_ provide `libc++.a` even when `!hostPlatform.isStatic`...

Reply:
Seems reasonable to me. I'll try tomorrow and see what hapoens.

Reply:
Done.